File: android/app/src/main/kotlin/app/gsmlg/caddyapp/CaddyLogStream.kt
package app.gsmlg.caddyapp

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class CaddyLogStream(messenger: BinaryMessenger) :
    EventChannel.StreamHandler {

    private val channel = EventChannel(messenger, "com.caddy_app/caddy/logs")
    private var eventSink: EventChannel.EventSink? = null

    init {
        channel.setStreamHandler(this)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    fun sendLog(message: String) {
        eventSink?.success(message)
    }

    fun dispose() {
        channel.setStreamHandler(null)
    }
}

File: ios/Runner/CaddyLogStream.swift
import Flutter
import UIKit

class CaddyLogStream: NSObject, FlutterStreamHandler {
    private let channel: FlutterEventChannel
    private var eventSink: FlutterEventSink?

    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterEventChannel(name: "com.caddy_app/caddy/logs", binaryMessenger: messenger)
        super.init()
        channel.setStreamHandler(self)
    }

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }

    func sendLog(_ message: String) {
        eventSink?(message)
    }

    func dispose() {
        channel.setStreamHandler(nil)
    }
}
